File: decompiled/sources/freemarker/template/TemplateDirectiveModel.java
package freemarker.template;

import freemarker.core.Environment;
import java.io.IOException;
import java.util.Map;

public interface TemplateDirectiveModel extends TemplateModel {
    void execute(Environment environment, Map map, TemplateModel[] templateModelArr, TemplateDirectiveBody templateDirectiveBody) throws TemplateException, IOException;
}

File: decompiled/sources/freemarker/template/TemplateDirectiveModel.kt
package freemarker.template

import freemarker.core.Environment
import java.io.IOException

interface TemplateDirectiveModel : TemplateModel {
    @Throws(TemplateException::class, IOException::class)
    fun execute(
        environment: Environment,
        params: Map<*, *>,
        loopVars: Array<TemplateModel?>,
        body: TemplateDirectiveBody?,
    )
}
